homework/index.js

```diff
@@ -60,35 +25,55 @@ async function main(url) {
     const select = createAndAppend('select', root, { class: 'select' });
     createAndAppend('option', select, { text: 'Choose your favorite repo' });
     let sorted = json.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()));
+
     sorted.forEach(repo => {
       const name = repo.name;
       createAndAppend('option', select, { text: name });
-      const wraper = createAndAppend('div', root, { class: 'wraper' });
-      const repoInfo = createAndAppend('div', wraper, { class: 'repoinfo' });
-      const contribs = createAndAppend('div', wraper, { class: 'contribs' });
-      select.addEventListener('change', evt => {
-        const selectedRepo = evt.target.value;
-        const repo = json.filter(r => r.name == selectedRepo)[0];
-        repoInfo.innerHTML = '';
-        contribs.innerHTML = '';
-        addInfo('Name: ', repo.name);
-        addInfo('Desciption: ', repo.description);
-        addInfo('Number of forks: ', repo.forks);
-        addInfo('Updated: ', repo.updated_at)
-        const contribsUrl = repo.contributors_url;
-        getContributorInformation(contribsUrl, contribs)
-      })
+    })
+    const wraper = createAndAppend('div', root, { class: 'wraper' });
+    const repoInfo = createAndAppend('div', wraper, { class: 'repoinfo' });
+    const contribs = createAndAppend('div', wraper, { class: 'contribs' });
 
+    select.addEventListener('change', evt => {
+      const selectedRepo = evt.target.value;
+      const repo = json.filter(r => r.name == selectedRepo)[0];
+      getContributorInformation();
+      repoInfo.innerHTML = '';
+      contribs.innerHTML = '';
+      const addInfo = (label, value) => {
+        const container = createAndAppend('div', repoInfo, { class: 'container' });
+        createAndAppend('span', container, { text: label });
+        createAndAppend('span', container, { text: value });
+      };
+      addInfo('Name: ', repo.name);
+      addInfo('Desciption: ', repo.description);
+      addInfo('Number of forks: ', repo.forks);
+      addInfo('Updated: ', repo.updated_at)
 
     })
+
   }
   catch (err) {
     const root = document.getElementById('root');
     createAndAppend('div', root, { text: err.message, class: 'alert-error' })
   }
 
+  async function getContributorInformation(data) {
+    try {
+      const contribs = await fetch(data.contributors_url);
+      const jsonData = await contribs.json();
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
 
-  window.onload = () => main(HYF_REPOS_URL);
-}
\ No newline at end of file
+      jsonData.forEach(contributor => {
+        createAndAppend('div', contribs, { text: contributor.login, class: 'contributor' })
```

Comment:
`contribs` here isn't what you expect. It's the promise from line 63 and not the dom-element from line 35, you need to get it from the dom again (like you do on line 35) and use it in this function.

Reply:
I changed this. but now the contrinutors_url is undefined!

Reply:
Not related! Have a look at how you call `getContributorInformation` and what arguments it requires

Reply:
I think I get your point. I forgot to put the argument 'data' in the getContributorInformation function. 
I should figure out what to put there !
```